src/main.zig

```diff
@@ -731,66 +811,142 @@ pub fn main() !u8 {
     defer allocator.free(shader_source);
 
     var using_custom_frame_rate = false;
-    var target_frame_rate = output.refresh_rate;
+    var custom_frame_rate: ?u32 = null;
     if (options.options.@"frame-rate") |frame_rate| {
         using_custom_frame_rate = true;
         if (frame_rate <= 0) {
             std.log.err("frame rate must be positive, got {}", .{frame_rate});
             return 1;
         }
-        target_frame_rate = @intCast(frame_rate);
+        custom_frame_rate = @intCast(frame_rate);
     }
 
-    const expected_frame_time_ns = @as(u64, std.time.ns_per_s) / target_frame_rate;
+    var papers = std.ArrayListUnmanaged(*Paper){};
+    defer {
+        for (papers.items) |paper| paper.destroy();
+        papers.deinit(allocator);
+    }
 
-    var global_attributes = GlobalAttributes.init();
-    defer global_attributes.deinit();
-    global_attributes.bind();
+    if (registry_listener.outputs.items.len == 0) {
+        std.log.err("no outputs available, cannot render", .{});
+        return 1;
+    }
 
-    const shader = try Shader.create(allocator, shader_source, .{ .width = surface.width, .height = surface.height }, target_frame_rate);
-    defer shader.destroy(allocator);
+    // Identify which outputs to use
+    var selected_outputs = std.ArrayListUnmanaged(*Output){};
+    defer selected_outputs.deinit(allocator);
 
-    var next_frame_time: u64 = 0;
-    var render_frame: bool = false;
+    if (options.options.output) |wanted_name| {
+        // Specific output requested
+        var found = false;
+        for (registry_listener.outputs.items) |output| {
+            try output.wait(display);
+            if (std.mem.eql(u8, output.name, wanted_name)) {
+                try selected_outputs.append(allocator, output);
+                found = true;
+                break;
+            }
+        }
+        if (!found) {
+            std.log.err("output with name {s} not found", .{wanted_name});
+            std.log.info("available outputs:", .{});
+            for (registry_listener.outputs.items) |output| {
+                std.log.info("- {s} ({}x{}, {}Hz)", .{
+                    output.name,
+                    output.width,
+                    output.height,
+                    @round(@as(f32, @floatFromInt(output.refresh_rate)) / 1000),
+                });
+            }
+            return 1;
+        }
```

Comment:
This option should be additive and parsed as described above. We can keep the default as "all outputs, all native".